### PR TITLE
REGEX add pcres_match and pcres_match_group to opensips-cli

### PR DESCRIPTION
**Pull request by Integration-IT**
**Summary**
Add pcres_match and pcres_match_group to opensips-cli.

**Details**
Test your regex on a fly whithout script implementation nor external tool.

**Solution**
Perform live test like reload, test string regex against the regular expression pcre_regex or by using the groups readed from the text file.

Exported MI function :
- mi_pcres_match (exported from pcre_match)
- mi_pcres_match_group (exported from pcre_match_group)

**Compatibility**
Compiled under 3.4

**Closing issues**
Feature request : https://github.com/OpenSIPS/opensips/issues/3112

### CODE REVIEW

File: modules/regex/regex_mod.c [hunk 1]
```diff
@@ -249,9 +257,9 @@ static int mod_init(void)
 
 	return 0;
 
-err:
-	free_shared_memory();
-	return -1;
+	err:
```
Comment (razvancrainea) [inline modules/regex/regex_mod.c:260]: Please revert this change, it is misleading.

Comment (Integration-IT) [reply modules/regex/regex_mod.c]: done.
